Content.Server/Anomaly/Effects/ProjectileAnomalySystem.cs

```diff
@@ -39,28 +40,30 @@ private void OnSupercritical(EntityUid uid, ProjectileAnomalyComponent component
     private void ShootProjectilesAtEntities(EntityUid uid, ProjectileAnomalyComponent component, float severity)
     {
         var xform = Transform(uid);
-        var projectilesShot = 0;
-        var range = component.ProjectileRange * severity;
-        var mobQuery = GetEntityQuery<MindComponent>();
+        var projectileCount = (int) MathF.Round(MathHelper.Lerp(component.MinProjectiles, component.MaxProjectiles, severity));
+        var mobQuery = GetEntityQuery<MobStateComponent>();
+        var inRange = _lookup.GetEntitiesInRange(uid, component.ProjectileRange * severity, LookupFlags.Dynamic).ToList();
+        _random.Shuffle(inRange);
 
-        foreach (var entity in _lookup.GetEntitiesInRange(uid, range, LookupFlags.Dynamic))
-        {
-            if (projectilesShot >= component.MaxProjectiles * severity)
-                return;
+        var priority = inRange.Where(mobQuery.HasComponent).ToList();
```

Comment:
LINQ stinky, just check it when iterating and you can cut down on allocations, plus it would need refactoring when I make entitylookup use callbacks instead.